src/goose/utils/autocomplete.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+
+from rich import print
+
+
+def is_autocomplete_installed(file: Path) -> bool:
+    if not file.exists():
+        print(f"[yellow]{file} does not exist, creating file")
+        with open(file, "w") as f:
+            f.write("")
+
+    # https://click.palletsprojects.com/en/8.1.x/shell-completion/#enabling-completion
+    if "_GOOSE_COMPLETE" in open(file).read():
+        print(f"auto-completion already installed in {file}")
+        return True
+    return False
+
+
+def setup_bash() -> None:
+    bashrc = Path("~/.bashrc").expanduser()
+    if is_autocomplete_installed(bashrc):
+        return
+
+    with open(bashrc, "a") as f:
+        f.write('eval "$(_GOOSE_COMPLETE=bash_source goose)"\n')
+    print(f"installed auto-completion to {bashrc}")
+
+
+def setup_fish() -> None:
+    completion_dir = Path("~/.config/fish/completions").expanduser()
+    if not completion_dir.exists():
+        completion_dir.mkdir(parents=True, exist_ok=True)
+
+    completion_file = completion_dir / "goose.fish"
+    if is_autocomplete_installed(completion_file):
+        return
+
+    with open(completion_file, "a") as f:
+        f.write("_GOOSE_COMPLETE=fish_source goose | source\n")
+    print(f"installed auto-completion to {completion_file}")
+
+
+def setup_zsh() -> None:
+    zshrc = Path("~/.zshrc").expanduser()
+    if is_autocomplete_installed(zshrc):
+        return
+
+    with open(zshrc, "a") as f:
+        f.write("autoload -U +X compinit && compinit\n")
+        f.write("autoload -U +X bashcompinit && bashcompinit\n")
+        f.write('eval "$(_GOOSE_COMPLETE=zsh_source goose)"\n')
```

Comment:
IMO this shouldn't write to files. For instance my `~/.zshrc` is a bootstrap for my dotfiles, i already have `autoload compinit` elsewhere in files. My preference would be for these to be printed to STDOUT as instructions and let me do it by hand.

If I run this, I will have to go into my files, rip this out and move into the right place. I think many people do maintain their own dotfiles.

Reply:
easy enough! added `generate-shell-completions --shell=<shell>` in 9e83ed9

Reply:
combined both commands into one so it's now `goose shell-completions --generate`